File: app/src/main/java/com/ducluanxutrieu/quanlynhanvien/MyFirebaseMessageService.java
package com.ducluanxutrieu.quanlynhanvien;

import com.ducluanxutrieu.quanlynhanvien.Activity.MainActivity;
import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

public class MyFirebaseMessageService extends FirebaseMessagingService {
    @Override
    public void onNewToken(String s) {
        super.onNewToken(s);

    }

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
        if (remoteMessage == null) return;
        if (remoteMessage.getData() != null) {
            String s1 = remoteMessage.getData().get("title");
            String s2 = remoteMessage.getData().get("message");
            if (s1.startsWith("@@")){
                s1 = s1.substring(2);
                s1 = "Ask for a off day from: " + s1;
            }
            hanldeNotification( s1, s2);
        }
    }

    @Override
    public void onMessageSent(String s) {
        super.onMessageSent(s);
    }

    @Override
    public void onSendError(String s, Exception e) {
        super.onSendError(s, e);
    }

    private void hanldeNotification(String title, String message) {
        PushNotificationManager.getInstance().generateNotification(title, message, MainActivity.class);
    }
}

File: app/src/main/java/com/ducluanxutrieu/quanlynhanvien/util/MyFirebaseMessageService.kt
package com.ducluanxutrieu.quanlynhanvien.util

import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.media.RingtoneManager
import android.os.Build
import androidx.core.app.NotificationCompat
import android.util.Log
import com.ducluanxutrieu.quanlynhanvien.R

import com.ducluanxutrieu.quanlynhanvien.activity.ChatsActivity
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage

class MyFirebaseMessageService : FirebaseMessagingService() {
    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        Log.i("uiduid", remoteMessage.data.toString())
        val `object` = remoteMessage.data
        Log.i("uiduid", `object`.toString())

        val s1 = remoteMessage.data["title"]
        val s2 = remoteMessage.data["message"]
        val uid = remoteMessage.data["uid"]
        Log.i("uiuid", s1 + "s2:" + s2 + "uid" + uid)

        sendNotification(s1, s2, uid)
    }

    private fun sendNotification(title: String?, name: String?, uid: String?) {
        val intent = Intent(this, ChatsActivity::class.java)
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
        intent.putExtra("uid", uid)
        intent.putExtra("name", name)
        val pendingIntent = PendingIntent.getActivity(this, 0 /* Request code */, intent,
                PendingIntent.FLAG_ONE_SHOT)

        val channelId = getString(R.string.default_notification_channel_id)
        val defaultSoundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)
        val notificationBuilder = NotificationCompat.Builder(this, channelId)
                .setSmallIcon(R.drawable.ic_app_icon)
                .setContentTitle(title)
                .setContentText(name)
                .setAutoCancel(true)
                .setSound(defaultSoundUri)
                .setContentIntent(pendingIntent)

        val notificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager

        // Since android Oreo notification channel is needed.
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val channel = NotificationChannel(channelId,
                    "Channel human readable title",
                    NotificationManager.IMPORTANCE_DEFAULT)
            notificationManager.createNotificationChannel(channel)
        }

        notificationManager.notify(0 /* ID of notification */, notificationBuilder.build())
    }
}
